ntlm.go

```diff
@@ -198,6 +217,84 @@ func ntlmSessionResponse(clientNonce [8]byte, serverChallenge [8]byte, password
 	return response(hash, passwordHash)
 }
 
+func ntlmHashNoPadding(val string) []byte {
+	hash := make([]byte, 16)
+	h := md4.New()
+	h.Write(utf16le(val))
+	h.Sum(hash[:0])
+
+	return hash
+}
+
+func hmacMD5(passwordHash, data []byte) []byte {
+	hmacEntity := hmac.New(md5.New, passwordHash)
+	hmacEntity.Write(data)
+
+	return hmacEntity.Sum(nil)
+}
+
+func getTargetInformationData(bytes []byte) (infoBytes []byte, target string, err error) {
+	type2MessageLength := len(bytes)
+	targetNameAllocated := binary.LittleEndian.Uint16(bytes[14:16])
+	targetNameOffset := binary.LittleEndian.Uint32(bytes[16:20])
+	if type2MessageLength < int(targetNameOffset + uint32(targetNameAllocated)) {
+		return nil, target, errorInvalidServerType2Message
```

Comment:
I'd prefer you put in a more specific error message here:
fmt.Errorf("mssql: NTLMv2 while parsing X message length %d too small for target offset %d", x, y)